### ISSUE TITLE
MailingAddress ZipCode required by EVSS, but not by vets-api

### ISSUE THREAD

**Issue by sethdarragile6** [department-of-veterans-affairs/va.gov-team#60645]
## Issue Description
It was discovered during the course of [526 submit discovery](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/57859) that you can submit form JSON to vets-api minus the zipCode field within the mailingAddress object- however, this will fail when submitting to EVSS because they require it.

```
     "mailingAddress": {
      "country": "USA",
      "addressLine1": "string",
      "addressLine2": "string",
      "addressLine3": "string",
      "city": "string",
      "state": "OR",
      needed to add --> "zipCode": "90210"
    }
```
Interestingly, many address fields within the full submission JSON are zipCode-optional (full sample at end of doc). Is there supposed to be some sort of zipcode lookup in the mix somewhere here? 
---
## Tasks
- [x] Determine if this is truly optional within the frontend flow (i.e. repro w/FE)
- [ ] Search in Sentry/Datadog to confirm that submissions might be failing because of this

## Acceptance Criteria
- [ ] Either petition EVSS to make this zipCode optional, or enforce on front-end
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)
- [ ]

**Comment by sethdarragile6** [comment]
full JSON POST to http://localhost:3000/v0/disability_compensation_form/submit_all_claim

```
{
  "form526": {
    "alternateNames": [
      {
        "first": "John",
        "middle": "Jonny",
        "last": "Johnson"
      }
    ],
    "atRiskHousingSituation": "losingHousing",
    "attachments": [
      {
        "name": "private_medical_record.pdf",
        "confirmationCode": "d44d6f52-2e85-43d4-a5a3-1d9cb4e482a1",
        "attachmentId": "L015"
      }
    ],
    "bankAccountNumber": "123456789",
    "bankAccountType": "Checking",
    "bankName": "string",
    "bankRoutingNumber": "123456789",
    "completedFormAttachments": [
      {
        "name": "private_medical_record.pdf",
        "confirmationCode": "d44d6f52-2e85-43d4-a5a3-1d9cb4e482a1",
        "attachmentId": "string"
      }
    ],
    "confinements": [
      {
        "from": "2019-10-10",
        "to": "2020-12-31"
      }
    ],
    "serviceInformation": {
        "servicePeriods": [
            {
                "serviceBranch": "Air Force",
                "dateRange": {
                    "from": "2020-01-01",
                    "to": "2023-09-17"
                }
            }
        ],
        "separationLocationName": "Chicago",
        "separationLocationCode": "98310"
    },
    "employmentRequestAttachments": [
      {
        "name": "private_medical_record.pdf",
        "confirmationCode": "d44d6f52-2e85-43d4-a5a3-1d9cb4e482a1",
        "attachmentId": "L115"
      }
    ],
    "form0781": {
      "remarks": "string",
      "additionalIncidentText": "string",
      "additionalSecondaryIncidentText": "string",
      "otherInformation": [
        "string"
      ],
      "incidents": [
        {
          "personalAssault": true,
          "medalsCitations": "string",
          "incidentDate": "2019-10-10",
          "incidentLocation": {
            "country": "USA",
            "city": "string",
            "state": "OR",
            "additionalDetails": "string"
          },
          "incidentDescription": "string",
          "unitAssigned": "string",
          "unitAssignedDates": {
            "from": "2019-10-10",
            "to": "2020-12-31"
          },
          "personsInvolved": [
            {
              "name": {
                "first": "string",
                "middle": "string",
                "last": "string"
              },
              "rank": "string",
              "injuryDeath": "killedInAction",
              "injuryDeathOther": "string",
              "injuryDeathDate": "2019-10-10",
              "unitAssigned": "string",
              "description": "string"
            }
          ],
          "sources": [
            {
              "name": "string",
              "address": {
                "country": "USA",
                "addressLine1": "string",
                "addressLine2": "string",
                "city": "Los Angeles",
                "state": "CA",
                "zipCode": "90210"
              }
            }
          ]
        }
      ]
    },
    "form4142": {
      "limitedConsent": "string",
      "providerFacility": [
        {
          "providerFacilityName": "string",
          "treatmentDateRange": [
            {
              "from": "2019-10-10",
              "to": "2020-12-31"
            }
          ],
          "providerFacilityAddress": {
            "street": "string",
            "street2": "string",
            "city": "string",
            "postalCode": "92021",
            "country": "USA",
            "state": "OR"
          }
        }
      ]
    },
    "form8940": {
      "unemployability": {
        "mostIncome": 0,
        "yearEarned": "string",
        "job": "string",
        "disabilityPreventingEmployment": "string",
        "underDoctorHopitalCarePast12M": true,
        "doctorProvidedCare": [
          {
            "name": "string",
            "address": {
              "country": "USA",
              "addressLine1": "string",
              "addressLine2": "string",
              "addressLine3": "string",
              "city": "string",
              "state": "OR"
            },
            "dates": "string"
          }
        ],
        "hospitalProvidedCare": [
          {
            "name": "string",
            "address": {
              "country": "USA",
              "addressLine1": "string",
              "addressLine2": "string",
              "addressLine3": "string",
              "city": "string",
              "state": "OR"
            },
            "dates": "string"
          }
        ],
        "disabilityAffectedEmploymentFullTimeDate": "2020-01-02",
        "lastWorkedFullTimeDate": "2020-10-10",
        "becameTooDisabledToWorkDate": "2020-10-10",
        "mostEarningsInAYear": "string",
        "yearOfMostEarnings": "string",
        "occupationDuringMostEarnings": "string",
        "previousEmployers": [
          {
            "name": "string",
            "employerAddress": {
              "country": "USA",
              "addressLine1": "string",
              "addressLine2": "string",
              "addressLine3": "string",
              "city": "string",
              "state": "OR"
            },
            "phone": "6196196199",
            "typeOfWork": "string",
            "hoursPerWeek": 0,
            "dates": {
              "from": "2019-10-01",
              "to": "2020-12-31"
            },
            "timeLostFromIllness": "string",
            "mostEarningsInAMonth": 0,
            "inBusiness": true
          }
        ],
        "disabilityPreventMilitaryDuties": true,
        "past12MonthsEarnedIncome": 0,
        "currentMonthlyEarnedIncome": 0,
        "leftLastJobDueToDisability": true,
        "leftLastJobDueToDisabilityRemarks": "string",
        "receiveExpectDisabilityRetirement": true,
        "receiveExpectWorkersCompensation": true,
        "attemptedToObtainEmploymentSinceUnemployability": true,
        "appliedEmployers": [
          {
            "name": "string",
            "address": {
              "country": "USA",
              "addressLine1": "string",
              "addressLine2": "string",
              "addressLine3": "string",
              "city": "string",
              "state": "OR"
            },
            "workType": "string",
            "date": "2020-10-10"
          }
        ],
        "education": "Some elementary school",
        "receivedOtherEducationTrainingPreUnemployability": true,
        "otherEducationTrainingPreUnemployability": [
          {
            "name": "string",
            "dates": {
              "from": "2019-10-10",
              "to": "2020-12-31"
            }
          }
        ],
        "remarks": "string"
      }
    },
    "forwardingAddress": {
      "country": "USA",
      "addressLine1": "string",
      "addressLine2": "string",
      "addressLine3": "string",
      "city": "string",
      "state": "OR",
      "zipCode": "92032",
      "effectiveDate": {
        "from": "2019-10-10",
        "to": "2020-12-31"
      }
    },
    "hasTrainingPay": true,
    "homelessHousingSituation": "shelter",
    "homelessOrAtRisk": "no",
    "homelessnessContact": {
      "name": "string",
      "phoneNumber": "6196196199"
    },
    "isVaEmployee": true,
    "mailingAddress": {
      "country": "USA",
      "addressLine1": "string",
      "addressLine2": "string",
      "addressLine3": "string",
      "city": "string",
      "state": "OR",
      "zipCode": "90210"
    },
    "mentalChanges": {
      "depression": true,
      "obsessive": true,
      "prescription": true,
      "substance": true,
      "hypervigilance": true,
      "agoraphobia": true,
      "fear": true,
      "other": true,
      "otherExplanation": "string",
      "noneApply": true
    },
    "militaryRetiredPayBranch": "Air Force",
    "needToLeaveHousing": true,
    "newPrimaryDisabilities": [
      {
        "condition": "string",
        "cause": "NEW",
        "classificationCode": "string",
        "primaryDescription": "string",
        "causedByDisability": "string",
        "causedByDisabilityDescription": "string",
        "specialIssues": ["PTSD/1"],
        "worsenedDescription": "string",
        "worsenedEffects": "string",
        "vaMistreatmentDescription": "string",
        "vaMistreatmentLocation": "string",
        "vaMistreatmentDate": "string"
      }
    ],
    "newSecondaryDisabilities": [
      {
        "condition": "string",
        "cause": "NEW",
        "classificationCode": "string",
        "primaryDescription": "string",
        "causedByDisability": "string",
        "causedByDisabilityDescription": "string",
        "specialIssues": ["PTSD/2"],
        "worsenedDescription": "string",
        "worsenedEffects": "string",
        "vaMistreatmentDescription": "string",
        "vaMistreatmentLocation": "string",
        "vaMistreatmentDate": "string"
      }
    ],
    "otherAtRiskHousing": "string",
    "otherHomelessHousing": "string",
    "phoneAndEmail": {
      "primaryPhone": "6196196199",
      "emailAddress": "string@sdfds.com"
    },
    "privateMedicalRecordAttachments": [
      {
        "name": "string",
        "confirmationCode": "string",
        "attachmentId": "L107"
      }
    ],
    "ratedDisabilities": [
      {
        "decision_code": "SVCCONNCTED",
        "decision_text": "Service Connected",
        "name": "Diabetes mellitus0",
        "effective_date": "2018-03-27T21:00:41.000+0000",
        "rated_disability_id": "0",
        "rating_decision_id": "63655",
        "rating_percentage": "100",
        "related_disability_date": "2018-03-27T21:00:41.000+0000",
        "disabilityActionType": "INCREASE",
        "special_issues": [
          {
            "code": "TRM",
            "name": "Personal Trauma PTSD"
          }
        ]
      }
    ],
    "secondaryAttachment": [
      {
        "name": "string",
        "confirmationCode": "string",
        "attachmentId": "L229"
      }
    ],
    "separationPayBranch": "VA Form 21-0781a - Statement in Support of Claim for PTSD Secondary to Personal Assault",
    "separationPayDate": "string",
    "standardClaim": false,
    "unemployabilityAttachments": [
      {
        "name": "string",
        "confirmationCode": "string",
        "attachmentId": "L149"
      }
    ],
    "vaTreatmentFacilities": [
      {
        "treatmentCenterName": "string",
        "treatmentDateRange": {
          "from": "2019-10-10",
          "to": "2020-12-31"
        },
        "treatmentCenterAddress": {
          "country": "USA",
          "city": "string",
          "state": "OR"
        },
        "treatedDisabilityNames": [
          "string"
        ]
      }
    ],
    "waiveRetirementPay": true,
    "waiveTrainingPay": true
  }
}
```

**Comment by alatempa24** [comment]
Closing this since we have a template now.

**Comment by micahaspyr** [comment]
@sethdarragile6 I think this is something Carbs can take over.  Any objections?

**Comment by navaeddie** [comment]
> @sethdarragile6 I think this is something Carbs can take over. Any objections?

I think I can speak on behalf of Seth that our team has no objections.

**Comment by NB28VT** [comment]
Verified: this validation **is in fact enforced by the front end** in the three places I checked:

1. Entering your address in the first place (/disability/file-disability-claim-form-21-526ez/contact-information)
![Screenshot 2023-06-23 at 10.40.50 AM.png](https://images.zenhubusercontent.com/647627f92b5e617343beedd1/8b50e9b5-33ff-46a9-bc3b-979ea22953db)

2. The "review and submit page" at the end of the form flow (disability/file-disability-claim-form-21-526ez/review-and-submit):

![Screenshot 2023-06-23 at 10.46.13 AM.png](https://images.zenhubusercontent.com/647627f92b5e617343beedd1/a01e6360-f745-4a49-a8f5-2f035ae53518)


3. I also checked a portion of the flow I was aware of where you need to enter the address of a medical facilityt hat holds your records (/disability/file-disability-claim-form-21-526ez/supporting-evidence/private-medical-records-release)
![Screenshot 2023-06-23 at 10.43.58 AM.png](https://images.zenhubusercontent.com/647627f92b5e617343beedd1/a1c690bf-54c3-425a-bbab-5e05b55027dc)

**Comment by NB28VT** [comment]
Next I plan to look through the [Site Map](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1686768383603/3f4415dd2ddb71bbd104ebefd92a2c7b553ad792?invited=true&sender=u846d384c8bd13e5d451c0747) for any other places it may be possible to submit a blank zip code field.

**Comment by NB28VT** [comment]
Regarding searching Sentry, I'll need to look at Sam's validation/errors investigations to determine where an error like that would occur.

Regardless, I strongly recommended we enforce this with a validation in the vets-api, which should be a very straightforward fix (update Rails validations for that submit endpoint and test).

**Comment by NB28VT** [comment]
Moving this ticket past the assessment phase as I've verified:
* On the front end, both places where the user can edit the zip code for their address (disability/file-disability-claim-form-21-526ez/contact-information and isability/file-disability-claim-form-21-526ez/review-and-submit) will not allow submitting the form if that field is blank
* The submission payload shared in full above by @sethdarragile6 indicates the zip code is included in the final form submission payload.

Regarding potential rejections from EVSS for missing zip codes, @SamStuckey indicated logged submission errors from them likely don't provide that level of detail, but I'll let him comment here whether he thinks that's worth diving into or not and we can set up a time to do that.

**IN SUMMARY: my recommendation is refactor the vets-api to validate a zip code is submitted for the mailing address regardless**


**Note: I did not investigate other parts of the flow where a ZIP code field is present (hospitals, places of employment etc.), but I did document their locations, as well as places where the user is prompted for an address without being asked for a ZIP code, in case these are causing rejections on the EVSS side. Not sure where I will share that info yet**

**Comment by RakshindaAslam** [comment]
81 out of ~ 16k failed submissions happened due to invalid zip codes submitted on mailing address

**Comment by austinjprice** [comment]
Possibility to implement VA Profile address validation service for all zip code and address related issues - Matt to submit foreign address issue and we will review if these can merge

**Comment by austinjprice** [comment]
Closing as this will be addressed in foreign address re-design @juliepedtke and first implemented by @steele-lm team
